src/sentry/search/events/datasets/discover.py

```diff
@@ -780,6 +780,62 @@ def function_converter(self) -> Mapping[str, SnQLFunction]:
                     default_result_type="number",
                     private=True,
                 ),
+                SnQLFunction(
+                    "fn_span_exclusive_time",
```

Comment:
not sure if this is the best way to do this as there's duplication in the resulting query:
```
SELECT
    max(arrayJoin(arrayFilter((x, y, z) -> ((y = 'resource.img') AND (z = 'dc03ec9fa0d49d13')), spans.exclusive_time_32 AS `_snuba_spans.exclusive_time`, spans.op AS `_snuba_spans.op`, arrayMap(x -> lower(hex(x)), spans.group) AS `_snuba_spans.group`))) AS _snuba_fn_span_exclusive_time__resource_img__dc03ec9fa0d49d13_max,
    min(arrayJoin(arrayFilter((x, y, z) -> ((y = 'resource.img') AND (z = 'dc03ec9fa0d49d13')), `_snuba_spans.exclusive_time`, `_snuba_spans.op`, `_snuba_spans.group`))) AS _snuba_fn_span_exclusive_time__resource_img__dc03ec9fa0d49d13_min
FROM transactions_local
PREWHERE (transaction_name AS _snuba_transaction) = '/organizations/:orgId/performance/summary/spans/:spanSlug/'
WHERE ((duration AS _snuba_duration) < 900000.) AND ((transaction_op AS _snuba_transaction_op) = 'pageload') AND (('transaction' AS _snuba_type) = 'transaction') AND ((finish_ts AS _snuba_finish_ts) >= toDateTime('2022-02-19T23:59:12', 'Universal')) AND (_snuba_finish_ts < toDateTime('2022-03-21T11:51:12', 'Universal')) AND ((project_id AS _snuba_project_id) IN [1])
LIMIT 0, 1
```
is this performant?

Reply:
the goal here is to find all valid exclusive times per transaction, then arrayJoin them and find a min or a max. my concern here is that i'm not reusing exclusive time array but rather create it twice